keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt

```diff
@@ -37,18 +36,12 @@ interface ArtifactRepository : PeriodicallyCheckedRepository<DeliveryArtifact> {
    * @return `true` if a new version is persisted, `false` if the specified version was already
    * known (in which case this method is a no-op).
    */
-  fun store(name: String, type: ArtifactType, version: String, status: ArtifactStatus?): Boolean
+  fun storeVersion(artifact: PublishedArtifact): Boolean
```

Comment:
We now use a model to store and retrieve published artifact versions (`PublishedArtifact`).